core/src/main/java/org/springframework/security/core/context/SecurityContextHolder.java

```diff
@@ -69,34 +71,41 @@
 	}
 
 	private static void initialize() {
```

Comment:
This should be called during context init so shouldn't be called concurrently, right?

Reply:
This is called whenever the strategy changes. For example, if `setContextHolderStrategy` were called concurrently, then `initialize()` would be called concurrently. This isn't the way the API is intended to be used, though.